examples/spmm/spmm_cuda.cc

```diff
@@ -787,7 +787,11 @@ class SpMM25D {
         , a_rowidx_to_colidx_(a_rowidx_to_colidx)
         , b_colidx_to_rowidx_(b_colidx_to_rowidx) {
       this->set_priomap([this](const Key<3> &ijk) { return this->prio(ijk); });  // map a key to an integral priority value
-
+      auto num_devices = ttg::device::num_devices();
+      this->set_devicemap(
+        [num_devices](const Key<3> &ijk){
+          return ((((uint64_t)ijk[0]) << 32) + ijk[1]) % num_devices;
```

Comment:
Is `ijk[0]` ever used in the modulo computation? For powers of 2 number of devices, this is equivalent to `ijk[0] << 32 | ijk[1] & X` where `X` is only ones on low bits for `log_2(num_devices)`, so only the bits of `ijk[1]` would matter, would they not?

Reply:
Good point. But maybe choosing either i or j (instead of mashing them together) is the better way anyway to avoid duplicating at least one matrix? I guess it ultimately depends on the shape of the matrix which one to choose.

Reply:
Usually, one tries to minimize the surface (amount of comms) while maximizing the volume (amount of flops). This leads to splitting the 3D workspace (i, j, k) into cubes and placing the GPUs on a 2D or 3D grid to maximize reuse. With the 2D approach, if we set `go = gq` such that `gp * gq == num_devices` (ideal case, of course... with `num_devices == 2`, I don't think we can do much better than 1D and just need to select which dimension we split), we should have the map map to ` (ijk[0] % gp) * gq + (ijk[1] % gq)` I think. Again with `gq == 1` and `gp == 2`, that's probably what you are doing already.